src/airbrakes.py

```diff
@@ -1,6 +1,196 @@
+
+    Note: If passing True closes Airbrakes, invert the value of 
+    step_direction when initializing Airbrakes object
+
+calibrate(): 
+  Sets __max_pot_val by reading potentiometer in the fully open state
+  and sets __min_pot_val by reading the potentiometer in the closed state
+
+  Returns nothing.
+
+  Must be called before launch (on the launch pad) to calibrate the system.
+
+deployBrakes():
+  Opens the brakes to the specifed percentage of fully open.
+
+  Returns value of potentiometer at final position.
+
+  Note: This uses the potentiometer for feedback so there will be a small
+  error associated with this. Adjust this error by changing max_error in
+  method body.
+
+
+"""
+class Airbrakes:
+  
+  # Set when calibrate() is called
+  __max_pot_val = None
+  __min_pot_val = None
+
+  def __init__(self,
+      stepper_pin,
+      direction_pin, 
+      direction: bool,
+      ads_pot_pin=0,
+      step_angle=1.8,
+      microsteps=16,
+      step_delay_micro_sec=75,
+      gear_box_ratio=14):
+
+    # GPIO pins
+    self.step_pin = stepper_pin
+    self.dir_pin  = direction_pin
+
+    # Stepper motor specs
+    self.stepper_motor = {
+      "direction": bool(direction),
+      "step_delay": step_delay_micro_sec/1000000.0,
+      "step_angle": step_angle,
+      "microsteps": microsteps,
+      "gear_ratio": gear_box_ratio
+    }
+   
+    # Max steps to go from fully closed to fully open with 1.25 safety factor 
+    # Division by 6 is because airbrakes only require 1/6 of a turn to open
+    self.__max_steps_to_open = (1.25
+                                * (self.stepper_motor["gear_ratio"]
+                                   *(360/self.stepper_motor["step_angle"])
+                                   * self.stepper_motor["microsteps"])
+                                /6)
+
+    GPIO.setmode(GPIO.BOARD)
+
+    # Set the pins as outputs
+    GPIO.setup(self.step_pin, GPIO.OUT)
+    GPIO.setup(self.dir_pin, GPIO.OUT)
+
+    # Initialize ADC (ADS1115) for reading the potentiometer 
+    self.i2c = busio.I2C(board.SCL, board.SDA)
+    self.ads = ADS.ADS1115(self.i2c)
+    self.potentiometer = AnalogIn(self.ads, ads_pot_pin)
+
+  def __singleStep(self, step_direction: bool):
+
+    # ^ XORs the direction (inverting it if motor_direction dictates it)
+    GPIO.output(self.dir_pin, step_direction ^ self.stepper_motor["direction"])
+
+    # Pulse step pin (raise to 5V, then lower to 0V) to step the motor
+    GPIO.output(self.step_pin, GPIO.HIGH)
+    sleep(self.step_delay)
+    GPIO.output(self.step_pin, GPIO.LOW)
+
+  def calibrate(self):
+    
+    # Fully open brakes and record pot value
+    for _ in range(self.__max_steps_to_open):
+      self._singleStep(True)
+    self.__max_pot_val = self.potentiometer.value
+
+    # Fully close brakes and record pot value
+    for _ in range(self.__max_steps_to_open):
+      self.__singleStep(False)
+    self.__min_pot_val = self.potentiometer.value
+
+  def deployBrakes(self, percent):
```

Comment:
percent is determined from cd right?

Reply:
Percent is based on the cd table lookup that the kalman filter outputs.